### DISCUSSION TITLE
Hello everybody!

How i can use async socketio client with proxy?

### DISCUSSION THREAD

**Discussion by Soulbadguy54** [miguelgrinberg/python-socketio#815] [Q&A]
Hello, in docs i found that request.session() object should be passed on AsyncClient() creation.

But this causes exception (Attribute error): session object do not has closed attribute.
Ofc, i didn't call it in my code, it was called in underlayer. 

Should i pass async session aiohttp session instead of request.session()?

Any example containing async client + proxy usage?

Thx for help!

**Comment by miguelgrinberg** [comment]
The `AsyncClient()` class takes a `aiohttp.ClientSession()` object for the HTTP session.

**Reply by Soulbadguy54** [reply]
Thank you :)

**Reply by heatht13** [reply]
Passing an aiohttp.ClientSession() to AsyncClient() using the http_session parameter makes sense, but how do you instantiate the ClientSession to utilize a proxy? Is setting trustenv to True and setting env vars the only way?

**Reply by heatht13** [reply]
my solution was to install aiohttp_proxy:
```
from aiohttp_proxy import ProxyConnector
import aiohttp

session = aiohttp.ClientSession(connector=ProxyConnector.from_url(PROXY_URL))
socketio_client = socketio.AsyncClient(http_session=session)
```
